services/wallet/controllers_v3_test.go

```diff
@@ -670,7 +670,12 @@ func TestLinkZebPayWalletV3_InvalidKyc(t *testing.T) {
 				},
 			})
 
-		s, _    = wallet.InitService(datastore, nil, nil, nil, nil, nil)
+		met = &mockMetric{
+			fnLinkFailureZP: func(cc string) {
+				assert.Equal(t, "IN", cc)
```

Comment:
Note: Generally I would not encourage assertions in mocks. Strictly speaking, the purpose of the surrounding test is to check the linking functionality, not the metrics. So probably we could avoid asserting in the mock.

However, if your intent is to double check, then I am fine with this. The same applies to the similar use below (in `TestLinkZebPayWalletV3`).

Reply:
generally a mock is used to verify the behaviour of the code under test by checking whether the correct methods were called and whether they were called with the correct arguments. In this case the mock is asserting that link failure has been called with the expected country code "IN". 

In the spirit of the test which is using mocking in the more traditional sense and in light of deprecating our use of gomock its doing the assertion.

Reply:
Thank you, that makes sense.